src/renderers/shared/reconciler/ReactCompositeComponent.js

```diff
@@ -630,7 +630,7 @@ var ReactCompositeComponentMixin = {
     if (this._pendingElement != null) {
       ReactReconciler.receiveComponent(
         this,
-        this._pendingElement || this._currentElement,
+        this._pendingElement,
```

Comment:
not sure if any of these are possible in real scenario  but this code could theoretically perform differently then before I believe.
<img width="299" alt="screen shot 2016-03-03 at 10 36 06 pm" src="https://cloud.githubusercontent.com/assets/8263298/13517325/5b4b2396-e190-11e5-9a79-b91b254e4f9f.png">